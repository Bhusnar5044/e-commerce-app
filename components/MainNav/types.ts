import { Category } from '@/types';

export interface IProps {
    data: Category[];
}
