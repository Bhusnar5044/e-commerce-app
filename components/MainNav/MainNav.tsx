'use client';

import Link from 'next/link';
import { usePathname } from 'next/navigation';
import { memo } from 'react';

import { cn } from '@utils';
import { Container } from '../ui/Container/Container';
import { IProps } from './types';

export const MainNav: React.FC<IProps> = memo(({ data }) => {
    const pathname = usePathname();

    const routes = data.map((route) => ({
        href: `/category/${route}`,
        label: route,
        active: pathname === `/category/${route}`,
    }));

    return (
        <Container>
            <nav className="mx-6 flex flex-wrap items-center justify-around">
                {routes.map((route) => (
                    <Link
                        key={route.href}
                        href={route.href}
                        className={cn(
                            'bg-blue-100 hover:bg-blue-200 text-blue-800 text-xs font-semibold mr-2 mb-2 px-2.5 py-0.5 rounded dark:bg-gray-700 dark:text-blue-400 border border-blue-400 inline-flex items-center justify-center',
                            route.active ? 'text-black' : 'text-neutral-500',
                        )}
                    >
                        {route.label}
                    </Link>
                ))}
            </nav>
        </Container>
    );
});

MainNav.displayName = 'MainNav';
