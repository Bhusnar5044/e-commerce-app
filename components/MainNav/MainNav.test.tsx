import { render } from '@test-utils';
import { MainNav } from './MainNav';

describe('MainNav', () => {
    it('should render properly', () => {
        const { container } = render(<MainNav data={[{ id: '1', name: 'dummy' }]} />);
        expect(container).toMatchSnapshot();
    });
});
