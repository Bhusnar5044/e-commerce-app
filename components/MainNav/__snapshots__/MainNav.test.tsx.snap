// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainNav should render properly 1`] = `
<div>
  <div
    class="mx-auto max-w-7xl"
  >
    <nav
      class="mx-6 flex flex-wrap items-center justify-around"
    >
      <a
        class="bg-blue-100 hover:bg-blue-200 text-xs font-semibold mr-2 mb-2 px-2.5 py-0.5 rounded dark:bg-gray-700 dark:text-blue-400 border border-blue-400 inline-flex items-center justify-center text-neutral-500"
        href="/category/dummy"
      >
        dummy
      </a>
    </nav>
  </div>
</div>
`;
